sqlglot/dialects/hive.py

```diff
@@ -51,6 +51,17 @@
 DIFF_MONTH_SWITCH = ("YEAR", "QUARTER", "MONTH")
 
 
+def constraint_sql(self, expression: exp.Constraint) -> str:
+    this = self.sql(expression, "this")
```

Comment:
you need to call copy up here, move this function into the generator directly and clean it up

Reply:
Copy the expression object? and 
in-line the code with the Generator list for more concise code?

Reply:
yea there’s no need to pass this in

Reply:
The `this` is the name of the constraint. Removing this removes the constraint name.

```python
def constraint_sql(self, expression: exp.Constraint) -> str:
    expression = expression.copy()
    this = self.sql(expression, "this")

    # remove unsupported properties
    expression.expressions.remove(expression.find(exp.Properties))
    expression.expressions.remove(expression.find(exp.OnProperty))

    expressions = self.expressions(expression, sep=" ", flat=True)
    return f"CONSTRAINT {this} {expressions}"
    
```